src/libpmemfile-posix/pmemfile-posix.c

```diff
@@ -78,25 +79,15 @@ libpmemfile_posix_init(void)
 
 	char *tmp = getenv("PMEMFILE_BLOCK_SIZE");
 	if (tmp) {
-		/*
-		 * XXX: I don't like parsing arbitrary input with
-		 * such functions.
-		 *
-		 * "...atoll need not affect the value of the integer
-		 * expression errno on an error. If the value of the result
-		 * cannot be represented, the behavior is undefined..."
-		 *
-		 * Also:
-		 * "An atoll is a ring-shaped coral reef, island, or series
-		 * of islets."
-		 */
-		long long tmpll = atoll(tmp);
-		if (tmpll < 0)
+		unsigned long long tmpll = strtoull(tmp, NULL, 0);
+		if (tmpll == ULLONG_MAX) {
```

Comment:
One should also check the endptr -- if it points to more characters, that means the input was partially parsed. This would parse `1024K` as 1024 without any warning, which might not be what the user expects.